packages/1155-contracts/src/delegation/ZoraCreator1155PremintExecutorImpl.sol

```diff
@@ -214,21 +214,28 @@ contract ZoraCreator1155PremintExecutorImpl is
         );
     }
 
-    /// @notice Returns the version of the premint signature that the contract supports
+    /// @notice Returns the versions of the premint signature that the contract supports
     /// @param contractAddress The address of the contract to check
-    /// @return The version of the premint signature that the contract supports.  If it doesn't support premint
-    /// returns 0
-    function supportedPremintSignatureVersion(address contractAddress) external view returns (string memory) {
+    /// @return versions The versions of the premint signature that the contract supports.  If contract hasn't been created yet,
+    /// assumes that when it will be created it will support the latest versions of the signatures, so the function returns all versions.
+    function supportedPremintSignatureVersions(address contractAddress) external view returns (string[] memory versions) {
+        // if contract hasn't been created yet, assume it will be created with the latest version
+        // and thus supports all versions of the signature
+        if (contractAddress.code.length == 0) {
+            return ZoraCreator1155Attribution.allVersions();
```

Comment:
should we return all versions here or just an array of 1 with the latest?

Reply:
good q. if the contract hasn't been created yet, and a premint v1 signature has been created for that address, then the new contract can be created with the v1 version, so in theory both v1 and v2 would work

Reply:
i do see how this is wierd that we are returning all versions if the contract hasn't been created yet.  may be just simpler to revert.  what do you think @iainnash @kulkarohan ?

Reply:
I don't mind all versions tbh, feels cleaner.